### PR TITLE
Write log entries of external nodes to console

### PR DESCRIPTION
**Pull request by brwe**
Before external nodes wrote logs only to a file which makes it tricky to debug
test failures.

As an alternative to this we also discussed to 
- set `inheritIO()` on the process that starts es (this actually prints the log to console too) and try to get the pid and port from the log file. We would then have to make sure than that each node name is only used once in each test run because otherwise we would have several pids and ports in each log file and would have to chose.
- set `inheritIO()` and write the pid to a file and then read pid from the file get port from the cluster.

This here just seemed most convenient to implement.

### CODE REVIEW

File: qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java [hunk 1]
```diff
@@ -309,7 +285,7 @@ private static void killAllBackwardsNodes() throws IOException, InterruptedExcep
             logger.info("Killing backwards nodes running at [{}]", pids);
         }
         for (String pid : pids) {
-            new ProcessInfo(null, pid).stop();
+            new ProcessInfo(null, pid, null).stop();
```
Comment (nik9000) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: Maybe stop should be a static method now?

Comment (nik9000) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: Maybe it always should have been....

Comment (brwe) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: I'd be happy to change but I do not understand why it should be static?

Comment (nik9000) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: Just so you don't have to build the object with so many nulls and throw it away again. Its no big deal.

Comment (brwe) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: sorry I am dumb. how would making it static help? we access other non static members in it.

File: qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java [hunk 2]
```diff
@@ -373,10 +349,14 @@ private void start(Deque<String> commandLine) throws IOException {
...
             String port = null;
             String pid = null;
-            BufferedReader stdout = null;
             try {
                 process = builder.start();
-                stdout = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
+                final PipedInputStream inputFromExternalNode = new PipedInputStream();
+                final PipedOutputStream outputFromExternalNode= new PipedOutputStream(inputFromExternalNode);
+                BufferedReader stdout = new BufferedReader(new InputStreamReader(inputFromExternalNode, StandardCharsets.UTF_8));
+                final AtomicBoolean writeToPipe = new AtomicBoolean(true);
```
Comment (nik9000) [inline qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java]: I remember this change now!

I wonder if you can drop the atomic boolean and the pipes and stuff and use a TeeOutputStream or something. This just feels complicated.

Maybe only start the LoggerThread after the process has started fully? Before that just read from stdout like we were before?

**Comment by brwe** [comment]
This seems messy and after a discussion with @nik9000 I decided to try the `inheritIO` anyway in #15290. I'll close this here once we made  decision.

**Pull request closed without merging**